symengine/sets.cpp

```diff
@@ -233,6 +233,46 @@ vec_basic Interval::get_args() const
     return {start_, end_, boolean(left_open_), boolean(right_open_)};
 }
 
+RCP<const Set> Reals::set_intersection(const RCP<const Set> &o) const
+{
+    return o;
```

Comment:
`o` could have complex values right?

Reply:
Thanks for pointing this out. I'll try to figure out how to do this. Ideas on how to handle complex numbers are welcome. I am thinking:

Interval, Empty - Must be real and can be handled as currently
FiniteSet - Can contain complex numbers. Could be handled directly
all others - Can contain complex numbers so create Intersection class